### PR TITLE
Create africa-interconnections.yaml

### PR DESCRIPTION
**Pull request by sandrinecharousset**
potential list of interconnections between countries based on the list of countries with common border

### CODE REVIEW

**Comment by MohElabbas** [comment]
@sandrinecharousset, there are a couple of things:
There are some double entries that should be eliminated:
 - Burkina Faso>Mali
 - Gambia>Senegal
 - Togo>Ghana

Some intercontinental interconnections should be included:
- Morocco>Spain
- Egypt>Jordan
- Egypt>Saudi Arabia (potential but yet to be constructed)

Some countries' names should be changed to the official recent ones:
- Swaziland should be changed to Eswatini
- Gambia should be The Gambia

Some countries are not included at all:
- Malawi (it is different from Mali)
- Equatorial Guinea (there are three Guinea-ian countries in Africa)

Finally, there are some listed interconnections that I think should be removed because they are not really "potential" (but it is up to you on this point):
- Angola>Congo & Congo>Angola: This is because Angola only borders Congo through its northern enclave of Cabinda, which is a small part that is not directly connected to the mainland of Angola. So, I wouldn't really include it.
- Libya>Sudan & Sudan>Libya: This border is really a huge desert and very far from the main grids of the two countries. There are even no close cities from both sides on this border.

**Comment by sandrinecharousset** [comment]
> @sandrinecharousset, there are a couple of things: There are some double entries that should be eliminated:
> 
>     * Burkina Faso>Mali
> 
>     * Gambia>Senegal
> 
>     * Togo>Ghana
> 
> 
> Some intercontinental interconnections should be included:
> 
>     * Morocco>Spain
> 
>     * Egypt>Jordan
> 
>     * Egypt>Saudi Arabia (potential but yet to be constructed)
> 
> 
> Some countries' names should be changed to the official recent ones:
> 
>     * Swaziland should be changed to Eswatini
> 
>     * Gambia should be The Gambia
> 
> 
> Some countries are not included at all:
> 
>     * Malawi (it is different from Mali)
> 
>     * Equatorial Guinea (there are three Guinea-ian countries in Africa)
> 
> 
> Finally, there are some listed interconnections that I think should be removed because they are not really "potential" (but it is up to you on this point):
> 
>     * Angola>Congo & Congo>Angola: This is because Angola only borders Congo through its northern enclave of Cabinda, which is a small part that is not directly connected to the mainland of Angola. So, I wouldn't really include it.
> 
>     * Libya>Sudan & Sudan>Libya: This border is really a huge desert and very far from the main grids of the two countries. There are even no close cities from both sides on this border.

@MohElabbas I did the changes as requested (thanks!), apart from changing name of Gambia as this is how it is named in the ISO we are following..... (I have absolutely no opinion about that, just want the whole stuff to be consistent.....)
Many thanks for your review!!

File: definitions/region/africa-interconnections.yaml [hunk 1]
```diff
@@ -0,0 +1,227 @@
+common_regions:
```
Comment (phackstock) [inline definitions/region/africa-interconnections.yaml:1]: The label common_regions is usually reserved for comparison regions between models. 
I would suggest something more obvious like:

```suggestion
Two country regions:
```

**Review by phackstock — COMMENTED** [review]
Just one quick comment from my side. 
@sandrinecharousset just to clarify, does your model report these two country regions? Because if not I'm not sure if we need to add them right now.

File: definitions/region/africa-interconnections.yaml
Comment (phackstock) [inline definitions/region/africa-interconnections.yaml:1]: You'll need to use a list format for this file. Meaning that `common_regions` needs to be a list entry like so `- common_regions` and all of the regions inside need to be indented by two spaces

**Comment by sandrinecharousset** [comment]
> Just one quick comment from my side. @sandrinecharousset just to clarify, does your model report these two country regions? Because if not I'm not sure if we need to add them right now.

@phackstock Yes!! Not only my model but nearly all models in the project. Interconnections are a central topic in this project as we have the power pools. And I can add that we don't only report results on these interco (such as the flows, the investments for increasing the intercos) but we also need data...... These directional regions were the main extension in the iamc dataformat that was made in open entrance.

**Comment by sandrinecharousset** [comment]
> > Just one quick comment from my side. @sandrinecharousset just to clarify, does your model report these two country regions? Because if not I'm not sure if we need to add them right now.
> 
> @phackstock Yes!! Not only my model but nearly all models in the project. Interconnections are a central topic in this project as we have the power pools. And I can add that we don't only report results on these interco (such as the flows, the investments for increasing the intercos) but we also need data...... These directional regions were the main extension in the iamc dataformat that was made in open entrance.

And I would add that we need to ensure consistency between these directional regions and the 3 regions from infrafair (all couples region>region in the infrafair list must be in the above list......)

**Review by phackstock — APPROVED** [review]
@sandrinecharousset, I took the liberty of fixing your file format. Now it works.
@MohElabbas, looks good to me but if there's more details to check I'll hold off on the merge.

**Comment by phackstock** [comment]
Perfect, thanks @MohElabbas for the super quick confirmation.

File: definitions/region/africa-interconnections.yaml
Comment (sandrinecharousset) [reply definitions/region/africa-interconnections.yaml:1]: > You'll need to use a list format for this file. Meaning that `common_regions` needs to be a list entry like so `- common_regions` and all of the regions inside need to be indented by two spaces

OK thanks, this wasn't absolutely clear to me , now I will not make the same mistake in the future :-)

**Comment by sandrinecharousset** [comment]
> @sandrinecharousset, I took the liberty of fixing your file format. Now it works. @MohElabbas, looks good to me but if there's more details to check I'll hold off on the merge.

@phackstock Thanks, all good for me too

**Pull request merged**